### DISCUSSION TITLE
Migrating images to a different directory

### DISCUSSION THREAD

**Discussion by hugohabicht01** [immich-app/immich#4436] [Q&A] [answered]
Hey there, recently I've setup this brilliant product on my server and uploaded all my pictures. However, my disk is now at 90% usage and it will soon be full. My solution to that would be to add a new drive, which is going to be dedicated to storing images. 
How do I safely migrate all the images which I have already uploaded to the new drive, without corrupting anything? Immich is currently running in a standard docker-compose config, as advised on the wiki. 

These are the steps I'm planning on taking:
1. stop the entire stack
2. mount the new drive
3. copy over the entire folder specified in `UPLOAD_LOCATION`
4. change the `UPLOAD_LOCATION` env variable to the new directory
5. start the immich stack again with `docker-compose up -d --force-recreate`
6. check whether all the images are still available via the web interface
7. delete the original folder

Is that the correct procedure? Am I missing something that might end up corrupting the setup?
Thanks

**Comment by bo0tzz** [comment, marked as answer]
That should work!

**Reply by tristankirkpatrick** [reply]
Did this work for you @hugohabicht01 ? I tried exactly this and it is as though the server isn't pointing at the new upload location, as it shows the same GB available. Also, the thumbnail links work, but when I click the photos, I get a broken image...

**Reply by hugohabicht01** [reply]
I have not fully done the migration yet, so far I've only added the new drive and set the file directory to the new drive. So far everything seems to be working fine, but I have not deleted the old drive yet. I'll do that within the next couple of days and let y'all know how it goes

**Reply by hugohabicht01** [reply]
apologies about the late update. I've just gotten rid of the original folder and everything is still working 100% fine. If you follow the steps which I outlined in my first post, everything should go smoothly

**Reply by leleogere** [reply]
> start the immich stack again with `docker-compose up -d --force-recreate`

Would you mind explaining why `--force-recreate` is needed?

**Reply by hugohabicht01** [reply]
I'm not entirely certain whether its actually needed or not, but it worked for me. The reason I added it, was to make sure that all the containers are fully fresh and don't potentially have some state stored internally (not sure whether thats the case or not, but it didn't hurt to add it)

**Reply by lusu007** [reply]
I successfully followed your instructions and all my images are available after moving the directory. However, the available disk space didn't change in the UI. 😅

**Reply by pikepa** [reply]
I successfully followed your instructions. Many thanks.

**Comment by kunzou** [comment]
I simply moved the whole library directory to my new drive and created a symbol link points to the new library path and it worked like a charm

**Reply by SuyashDahale13** [reply]
> I simply moved the whole library directory to my new drive and created a symbol link points to the new library path and it worked like a charm

i am new to home server and linux, can you please share how did you do that... commands and steps. it will be very healpful.

**Reply by arLevi** [reply]
a sym link is created by `ln -s </full/path/old dir> </full/path/new_dir>`

**Comment by arLevi** [comment]
After the following answer's instructions, on windows it now gives me the error which makes the docker immich-server restarting non stop:
```
ERROR [Microservices:StorageService] Failed to read upload/encoded-video/.immich: Error: ENOENT: no such file or directory, open 'upload/encoded-video/.immich'
```
And nothing from this URL below works ...
https://immich.app/docs/administration/system-integrity/#folder-checks

**Reply by arLevi** [reply]
Just commenting that i'm using `Windows` & `Docker Desktop`. for some reason the `Docker Desktop` did not recognized my new hard-drive ( USB-C ) even when I created a volumn mapping etc ... I noticed it when I tried the "backup & restore" documentation and the `*.sql` files were never shared in the new drive.

Eventually, I restarted `Docker desktop` - then i noticed the files were there, and then `Immich` worked in the new directory w/o doing anything ... 

it was all comes down to - the docker never "saw" the disk ... got nothing to do (in my case) to a missing`.immich` file etc ...

**Comment by markmeplease** [comment]
Following the instruction did not work for me, when I start running the container, it is creating the directories in the old drive. Without luck, I tried to transfer the "Docker Disk Image Location" to the new drive then execute the "`docker compose up -d --force-recreate`" and it worked just like a charm. I also think that the flag `--force-recreate` is not necessary, but as the author said, it does not hurt, so I still execute the flag 🤣🤣🤣

This is what I did for reference:
1. Copy and paste the "immich-app" directory from the old drive to the new drive
2. Change the **Disk image location** of the docker to the new drive using the UI under the Docker _Settings > Resources > Advanced > Disk image location_
3. Execute the command `docker-compose up -d --force-recreate`
4. Done! 🎉🎊🍻

Note: I am using Docker Desktop on Windows WSL2

Storage Capacity was also updated in the Immich UI

![image](https://github.com/user-attachments/assets/5417c72c-f715-4aa6-a6b5-c9c062f0a393)

DIrectory

![image](https://github.com/user-attachments/assets/5dbff3ac-4372-4620-bebe-53a912aca13f)

**Comment by sprinkles5** [comment]
Just used this to migrate my immich library from an external drive to an internal one on an Ubuntu 24.04 server and it worked beautifully. I was intimidated and confused by the backup and migration instructions on the immich website, so I'm glad I found this. Thank you!

**Comment by SavanFlou** [comment]
I am trying to do this but my Immich is on Truenas Scale. 
Do you know how to adapt this method?
